services/report/tests/unit/test_process.py

```diff
@@ -1229,3 +1204,52 @@ def test_process_raw_upload_with_carryforwarded_flags(self):
                 (1, 1, None, [[session.id, 1, None, None, None]], None, None)
             ],
         }
+
+    def test_process_raw_upload_carryforward_flag_empty_upload(self):
+        """Tests that an upload with carryforward flag doesn't create a new session
+        in the original report for the flag it was uploaded with.
+
+        That is, a new session is created in the commit's report only if the upload
+        was properly processed
+        """
+        original_report = self._generate_sample_report()
+        original_sessions = original_report.sessions
+        expected_sessions = {
```

Comment:
In the previous behaviour, would we have seen 3 sessions instead of 2 here, with two of them being for the flag "somethingold"?

Reply:
Yes. You can see a similar behavior in the test I altered. The removed parts are the outcome of the old behavior, where an extra session would be present.

Reply:
ah I see now, thanks for pointing that out